### ISSUE TITLE
AzureCluster failureDomains do not always match specific control plane AzureMachine SKU

### ISSUE THREAD

**Issue by nojnhuh** [kubernetes-sigs/cluster-api-provider-azure#5033]
/kind bug

**What steps did you take and what happened:**

CAPZ's AzureCluster controller contains logic to set the `status.failureDomains` to the available availability zones for the region of the cluster resources:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/89efbf34282c373765819e5c3c4a7e3bff2d20de/azure/services/resourceskus/cache.go#L170

This function loops through all VM SKUs, returning any availability zone without restrictions for at least one SKU to be set in the AzureCluster's failureDomains. But when a specific VM SKU in a region has a `Zone` restriction preventing VMs from being created in specific zones, CAPZ may still list those zones in its `status.failureDomains` if some other SKU in the same region does not have the same restrictions, causing VMs to be created in zones with restrictions and fail.

**What did you expect to happen:**

The AzureCluster failureDomains match the availability of the specific SKU of control plane VM being used.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by mboersma** [comment]
@nojnhuh should this be in the next milestone?

**Comment by nojnhuh** [comment]
I don't think this is something we need to do soon, at least since we've ironed out the regions we use in CI to avoid this.
